packages/adapter-drizzle-mysql/src/index.ts

```diff
@@ -0,0 +1,293 @@
+ *  sessionToken: text("sessionToken").notNull().primaryKey(),
+ *  expires: integer("expires").notNull(),
+ * })
+ *
+ * export const verificationTokens = pgTable("verificationToken", {
+ *  identifier: text("identifier").notNull(),
+ *  token: text("token").notNull(),
+ *  expires: integer("expires").notNull()
+ * }, (vt) => ({
+ *   _: primaryKey(vt.identifier, vt.token)
+ * }))
+ *
+ * const pool = new Pool({
+ *   connectionString: "YOUR_CONNECTION_STRING"
+ * });
+ *
+ * export const db = drizzle(pool);
+ *
+ * migrate(db, { migrationsFolder: "./drizzle" })
+ *
+ * ```
+ *
+ **/
+export function DrizzleAdapterMySQL(client: DbClient, {
+  users, sessions, verificationTokens, accounts
+}: Schema): Adapter {
+  return {
+    createUser: async (data) => {
+      const id = uuid()
+
+      await client
+        .insert(users)
+        .values({ ...data, id })
+
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.id, id))
+        .then(res => res[0])
+
+    },
+    getUser: async (data) => {
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.id, data))
+        .then(res => res[0])
+        ?? null
+    },
+    getUserByEmail: async (data) => {
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.email, data))
+        .then(res => res[0])
+        ?? null
+    },
+    createSession: async (data) => {
+      await client
+        .insert(sessions)
+        .values(data)
+
+      return client
+        .select()
+        .from(sessions)
+        .where(
+          eq(
+            sessions.sessionToken,
+            data.sessionToken
+          )
+        )
+        .then(res => res[0])
+    },
+    getSessionAndUser: async (data) => {
+      return client.select({
+        session: sessions,
+        user: users
+      })
+        .from(sessions)
+        .where(eq(sessions.sessionToken, data))
+        .innerJoin(users, eq(users.id, sessions.userId))
+        .then(res => res[0])
+        ?? null
+    },
+    updateUser: async (data) => {
+      if (!data.id) {
+        throw new Error("No user id.")
+      }
+
+      await client
+        .update(users)
+        .set(data)
+        .where(eq(users.id, data.id))
+
+      return client
+        .select()
+        .from(users)
+        .where(
+          eq(
+            users.id,
+            data.id
+          )
+        )
+        .then(res => res[0])
+
+
+    },
+    updateSession: async (data) => {
+      await client
+        .update(sessions)
+        .set(data)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+
+      return client
+        .select()
+        .from(sessions)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+        .then(res => res[0])
+    },
+    linkAccount: async (rawAccount) => {
+      await client
+        .insert(accounts)
+        .values(rawAccount)
```

Comment:
In the `accounts` schema: `expires_at`. But the rawAccount provided by the Google provider has `expires_in`. This will cause the insert to fail.
You'll probably want to pick out `expires_in` and add it to `expires_at: (Date.now()/1000 + expires_in)>>0`

Reply:
Interesting, I was under the impression that the Github provider would handle this at its layer. 🤔  I'll be sure to be cognizant of this while I go around and test!

Reply:
Going through testing now and it does look like this is working! Let me know if it isn't.